Civi/Core/Container.php

```diff
@@ -366,6 +366,8 @@ public function createEventDispatcher() {
     $dispatcher->addListener('hook_civicrm_coreResourceList', ['\CRM_Utils_System', 'appendCoreResources']);
     $dispatcher->addListener('hook_civicrm_getAssetUrl', ['\CRM_Utils_System', 'alterAssetUrl']);
     $dispatcher->addListener('hook_civicrm_alterExternUrl', ['\CRM_Utils_System', 'migrateExternUrl'], 1000);
+    $dispatcher->addListener('hook_civicrm_permissionList', ['\Civi\Api4\Action\Permission\Get', 'buildDefaultPermissions'], 1000);
```

Comment:
The API file strikes me as a strange place to put this callback. Normally business logic like this would go in a BAO or some other place maybe in a `\Civi\Core\Permission` class?

Reply:
Yeah, having it this way is kinda optimal from a file-loading (and code-reading) perspective. OTOH, the method should really be `private`, but it has to be `public` to work with the dispatcher, and that goes against the expectation where `public` methods in `Civi\Api4\Action\**` are part of the API surface.

I guess a standalone file would be OK as long as we sprinkle in some `@see`s for readability.

Reply:
@colemanw OK, I pulled it out.